sansa-ml-spark/src/main/scala/net/sansa_stack/ml/spark/entity_resolution/Commons.scala

```diff
@@ -16,17 +15,22 @@ import org.apache.spark.RangePartitioner
 import org.apache.spark.storage.StorageLevel
 import org.apache.spark.RangePartitioner
 import org.apache.spark.RangePartitioner
+import scala.io.Source
+import java.io.Serializable
 
 /* A generic Entity Resolution Approach for larger datasets(for e.g. 24.8GB)
  * Dataset1 has 47 million entities, approx 19GB
  * Dataset2 has 19 million entities, approx 5.8GB
  * */
-class EntityResolution_RDFData_CountVectorizer(spark: SparkSession, triplesRDD1: RDD[Triple], triplesRDD2: RDD[Triple],
-                                         teacher: DataFrame, threshold_subject: Double, jsimilarity_predicate: Double,
-                                         threshold_object: Double, vocab_size: Long, output_path: String) extends Serializable {
+abstract class Commons(val spark: SparkSession, val sourceData1: RDD[Triple], val sourceData2: RDD[Triple],
+                                         val teacher: DataFrame, val thresholdSubject: Double, val jsimilarityPredicate: Double,
+                                         val thresholdObject: Double, val vocabSize: Long, val outputPath: String) extends Serializable {
+
+@transient lazy val consoleLog: Logger = Logger.getLogger(getClass.getName)
+private val PARTITIONS = 400
```

Comment:
Even, though you considered my comment to make it a variable, but again this is quite static. If you want to get rid of such variables (which anyway are quite static), you could consider adding an `application.conf` file and add such variables there which can be tunned by the user. Please, resolve this one as well.

Reply:
Done.